File: PetHelper.Domain/Agendamento.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace PetHelper.Domain;

public class Agendamento : IEntidadeBase
{
    public int Id { get; set; }
    public int PetShopId { get; set; }
    public int ClienteId { get; set; }
    public Cliente? Cliente { get; set; }
    public int ServicoId { get; set; }
    public Servico? Servico { get; set; }
    public int PetId { get; set; }

    [NotMapped]
    public Pet? Pet { get; set; }

    public string HorarioMarcado { get; set; } = string.Empty;
    public DateTime DataAgendamento { get; set; }
    public DateTime DataCriacao { get; set; }
    public DateTime DataAtualizacao { get; set; }
}

File: PetHelper-App/src/app/models/agendamento.ts
import { Cliente } from "./cliente";
import { Pet } from "./pet";
import { Servico } from "./servico";

export interface Agendamento {
  id: number;
  petShopId: number;
  clienteId: number;
  cliente: Cliente;
  servicoId: number;
  servico: Servico;
  petId: number;
  pet: Pet;
  horarioMarcado: string;
  dataAgendamento: Date;
}
